README
CGI::Application::Plugin::Authentication

adds the ability to authenticate users in your CGI::Application modules.


INSTALLATION

To install this module, run the following commands:

    perl Build.PL
    ./Build
    ./Build test
    ./Build install


Alternatively, if you do not have Module::Build but you do have 'make',
you can used the included Makefile.PL and run the following commands:

    perl Makefile.PL
    make
    make test
    make install


DEPENDENCIES

CGI::Application >= 4.0
Scalar::Util
Class::ISA
Test::More
Test::Exception
Attribute::Handlers
UNIVERSAL::require

RECOMMENDED MODULES

Some extra modules are required for some of the features of this plugin

CGI::Application::Plugin::Authentication::Store::Session requires:
    CGI::Application::Plugin::Session


CGI::Application::Plugin::Authentication::Store::Cookie requires:
    CGI::Cookie
    MIME::Base64
    Digest::SHA1


COPYRIGHT AND LICENCE

Copyright (C) 2005, SiteSuite

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.



Changes
Revision history for {{$dist->name}}

{{$NEXT}}

0.23      2018-03-05 13:15:54-06:00 America/Chicago
   Support SQLite 3.22.0 in tests (thanks to Gregor Herrmann and Debian Perl)

0.22      2018-02-07 14:24:30-06:00 America/Chicago
   Fix failing tests on Win32

0.21      2017-03-15 22:54:23-05:00 America/Chicago
    Fix list-context calls to CGI::param().
    Fix failing tests with newer CGI versions.

0.20 Fri 29 Apr 2011
    Use Test::ConsistentVersion rather than Test::CheckVersion
    Fixed spelling errors and added descriptions - Closes #rt63839
    Removed now unused login_styles method from core module
    Added more tests completing test coverage for the Cookie module.
    Changed from using the Digest::SHA1 module to usuing Digest::SHA
    instead - Closes #rt67840

0.19 Wed 24 Nov 2010
    Incoporated developer changes. Made pod tests only run when TEST_AUTHOR=1.
    Automatically generate README.

0.18_2  Mon 20 Jul 2010
    Separate out display code and provide Basic and Classic alternatives to
    the login form. Added more examples. Implementd typo fix reported by
    Alexandr Ciornii. Added copyright declaration for icons in the login form.

0.18_1 Thu Jun 25 2010
    Put OS dependent guards in certain tests.
    Removed deprecated *_DESTINATION parameters.
    Added first test for devpopup and fixed undefined variable therein.
    Cannot use Test::More 0.92 as that has an issue with subtests.

0.18 Mon Jun 23 2010
    Completed test coverage for Driver and below
    TABLES parameter was not made mandatory in DBI driver.
    Fixed dependency on Test::More - must be at least version 0.90.
    Fixed handling of custom filters - now they must have a $param and a $value.
    Incorporated Christian Walde's suggested changes for compatibility with
    CGI::Application::Plugin::ActionDispatch.
    Adding tests around the HTML generated by the login_box function and
    verifying that it is taint free.
    Upped requiredment on CGI to 3.16 as that seems to cause unusual behaviour in returning
    self referring urls and further upped requirement on CGI to 3.16 as 3.15 has an
    unusual  interpretation of redirection headers.
    Fixed regular expressions in destination test to respect HTTP header CRLF conventions.

0.17  Thu Jan 21 19:16 GMT 2010
    Bug Fixes
    rt53533 - During initialization deferred checking the users credentials to the last possible moment.
    This ensures that sessions are not created unless actually required.
    Also attempted to document which functions add session/cookie state.

    Build Fixes
    rt35030 - Upped the requirement on Apache::Htpasswd to 1.8 as this simplifies our dependency management.

0.16  Mon Jan 18 22:18 GMT 2010
    Release dedicated to improving CPAN readiness - see RT:50670.
    Fixed spelling mistakes and added a test script that uses Test::Spelling.
    Fixed dependency on Test::Exception and added a test script that uses Test::Prereq::Build.
    Added test scripts to validate the MANIFEST file and Changes file.
    Removed files that are generated during './Build dist' from repository.

0.15  Tue Jan 12 15:42 GMT 2010
    Redist to fix tar format issues RT:45155 & 23705

0.14  Sun Oct 18 02:48:10 EST 2009
    Bug Fixes
    - Make sure the 'action' attribute of the login_box includes the
      path_info in the URL (resolves RT:38049 patch by Alex Becker)

    New Features
    - Add options for LIMIT and ORDER_BY to DBI driver (requested by tbone)
    - Add FORM_SUBMIT_METHOD option to the login box which allows you to
      change the form method to GET instead of the default POST.
      (resolves RT:34198 requested by jaldhar)
    - Add hooks for devpopup (resolves RT:35989 patch by Alexandr Ciornii)

0.13  Mon Mar 30 23:58:25 EST 2009
    Build fixes
    - There are no code changes to this release, just fixes in the build and
      test systems to solve some outstanding test failures
    - unit tests now skip tests where optional modules are not installed
      (gtermars@cpan.org)
    - remove build dependency on Test::Exception, replacing it with hand-rolled
      equivalent methods.  Newer Perls have been seen to fail our tests with
      "Bizarre copy of HASH in sassign..." when using Test::Exception.  Cause
      of the issue may be far lower/internal than Test::Exception, but
      replacing these methods eliminates the issue for us.  (gtermars@cpan.org)
    - cleanup prerequisites and build files (Alexandr Ciornii)

0.12

    Bug Fixes
    - when issuing a redirect, remember to set header_type to 'redirect'
      (reported by Graham TerMarsch)

    Other updates
    - Fix inconsistency in doc example (reported by POLETTIX)

0.11  Wed Jul 26 11:16:12 EDT 2006

    - suppress warning about missing Color::Calc unless the user is actually
      trying to use it (reported by Ron Savage)
    - explicitly load CGI.pm since some users may be using CGI::Simple
      (reported by Ron Savage)
    - Fixed bug in Cookie store where the expiry time of the cookie was ignored
      (patch by POLETTIX)

0.10  Thu May 18 22:59:56 EDT 2006

    - Add support for Authen::Simple (all Authen::Simple modules can
      be used directly as Drivers)
    - Made the login page much more customizable:
      - change any of the text
      - customize the colours - provide one base colour and lighter and
        darker shades are automatically generated (requires Color::Calc)
      - offer to remember the users username in a cookie for the next
        time they login
      - add option for a 'forgotten password' URL
      - add option for a 'register new account' URL
      - add option to supress the stylesheet so you can provide your own

0.09  Thu Jan 26 20:56:29 EST 2006

    API Changes
    - Added option to provide your own HTML for the login box through a callback
      (Shawn Sorichetti)
    - Added POST_LOGIN_RUNMODE callback (Shawn Sorichetti)

0.08  Wed Nov 23 20:20:24 EST 2005

    API Changes

Build.PL
use Module::Build;

Module::Build->new(
    module_name => 'CGI::Application::Plugin::Authentication',
    license     => 'perl',
    requires    => {
        'CGI::Application'    => 4,
        'Scalar::Util'        => 0,
        'Class::ISA'          => 0,
        'Attribute::Handlers' => 0,
        'UNIVERSAL::require'  => 0,
        'CGI'                 => '3.16',
        'MIME::Base64'        => 0,
        'Digest::SHA'         => 0,
        'perl'                => 5.006,
    },
    recommends => {
        'Digest::MD5'                       => 0,
        'CGI::Application::Plugin::Session' => 0,
        'Apache::Htpasswd'                  => '1.8',
        'Color::Calc'                       => '0.12',
    },
    build_requires => {
        'Test::More' => '0.93',
        'Test::Warn' =>
          0.11,    # older versions may have problems with fresh Sub::Uplevel
        'Test::MockObject' => 0,
        'Test::Exception'  => 0,
        'Test::NoWarnings' => 0,
        'Test::Regression' => 0, # login_box generates a lot of HTML to verify
        'Test::Taint'      => 0,
        'Test::Without::Module'      => 0,
        'Readonly' => 0,
    },
    meta_merge => {
        resources => {
            repository =>
'http://github.com/cees/cgi-application-plugin-authentication/tree',
        },
        keywords => [ 'cgiapp', 'Titanium', 'authentication' ],
    },
    create_readme=>1,
    create_makefile_pl => 'traditional',
    dist_author        => 'Cees Hek <ceeshek@gmail.com>',
)->create_build_script;

